src/axom/primal/operators/detail/intersect_bezier_impl.hpp

```diff
@@ -471,31 +471,39 @@ bool intersect_2d_circle_line(const Sphere<T, 2> &circ,
 
   T disc = circ.getRadius() * circ.getRadius() * dr * dr - D * D;
 
-  // Identify near-tangent cases
-  if(disc <= EPS * EPS)
+  // Identify near-tangent cases by computing the distance from the line to the circle center,
+  //  which should be equal to the radius of the circle if the line is tangent.
+  if(axom::utilities::isNearlyEqual(std::abs(D) / dr, circ.getRadius(), EPS))
   {
-    T ct;
-    Segment<T, 2> seg(a, b);
+    // Find the closest point on the line to the circle center
+    Point<T, 2> cp(a);
+    double num = (circ.getCenter()[0] - a[0]) * dx + (circ.getCenter()[1] - a[1]) * dy;
 
-    // Because the line is known to be tangent, there's only one intersection point,
-    //  and it would have to be at the closest point on the line to the circle center.
-    Point<T, 2> cp = closest_point(circ.getCenter(), seg, &ct);
-    T dist = primal::squared_distance(cp, circ.getCenter());
+    cp.array()[0] += num * dx / (dr * dr);
+    cp.array()[1] += num * dy / (dr * dr);
```

Comment:
Minor:  I don't think you need `array()` here since `T` is `double`

Does this work?
```suggestion
    cp[0] += num * dx / (dr * dr);
    cp[1] += num * dy / (dr * dr);
```

Reply:
Oh, whoops! You're totally right, reflex for in-place point addition, I suppose.